impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Abstract.js

```diff
@@ -71,48 +73,105 @@ define([
+        color: model.labelColor,
+        fontSize: model.labelSize || 12,
+        fontFamily: fontFamily === "Default" ? font.substring(font.indexOf(" ") + 1) : fontFamily
       };
+    },
 
-      options.series.forEach(function(row) {
-        row.label = label;
-      });
+    _legendOrientation: function(position) {
+      switch (position) {
+        case "left":
+        case "right":
+          return "vertical";
+        case "top":
+        case "bottom":
+          return "horizontal";
+        default:
+          return "horizontal";
+      }
     },
 
     _configureLegend: function(options, records) {
       var categories = [];
+      var model = this.model;
       var font = util.getDefaultFont(null, 14);
 
+      var top = "auto";
+      var left = "auto";
+      var position = model.legendPosition;
+      var orient = this._legendOrientation(position);
+      var fontWeight = model.legendStyle;
+      var fontFamily = model.legendFontFamily;
+      var legendBgColor = model.legendBackgroundColor;
+
+      if(position === "top" || position === "bottom") {
+        top = position;
+      } else {
+        left = position;
+      }
+      
       records.forEach(function(record) {
         categories.push(record.name);
       });
 
       options.legend = {
-        show: categories.length <= 1 ? false : true,
+        show: model.showLegend && categories.length > 1,
         data: categories,
         type: "scroll",
-        align: "auto",
+        align: "left",
+        icon: "circle",
+        orient: orient,
+        left: left,
+        top: top,
+        backgroundColor: legendBgColor.toLowerCase() !== "#ffffff" ? legendBgColor : "transparent",
+        itemWidth: 8,
+        itemHeight: 8,
         padding: 10,
         textStyle: {
-          fontSize: font.substring(0, font.indexOf(" ")),
-          fontFamily: font.substring(font.indexOf(" ") + 1)
+          fontWeight: fontWeight === "plain" ? "normal" : fontWeight,
+          color: model.legendColor,
+          fontSize: model.legendSize,
+          fontFamily: fontFamily === "Default" ? font.substring(font.indexOf(" ") + 1) : fontFamily
         }
       };
     },
 
-    _configureData: function(options, data) {
-      options.series.forEach(function(seriesEntry) {
-        seriesEntry.data = data;
-      });
+    _buildBgColor: function() {
+      var model = this.model;
+      var bgFillType = model.backgroundFill;
+      var bgColorFill = model.backgroundColor;
+      var bgColorEnd = model.backgroundColorEnd;
+
+      if(bgFillType !== "none") {
+        if(bgFillType === "solid") {
+          return bgColorFill;
+        } else {
+          return {
+            type: "linear",
+            x: 0, y: 0,
+            x2: 0, y2: 1,
+            colorStops: [
+              {offset: 0, color: bgColorEnd},
+              {offset: 1, color: bgColorFill}
+            ],
+          };
+        }
+      } else {
+        return "transparent";
+      }
```

Comment:
No need to do.
My preference for code style is to test, return, test return. No elses.
For example:
```js
if(bgFillType === "none") {
  return "transparent";
}

if(bgFillType === "solid") {
  return bgColorFill;
}

var bgColorFill = model.backgroundColor;
// ...
return { ... };
```

Reply:
Addressed.